core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -72,6 +74,10 @@ public class HTTPClient implements RESTClient {
   static final String CLIENT_GIT_COMMIT_SHORT_HEADER = "X-Client-Git-Commit-Short";
 
   private static final String REST_MAX_RETRIES = "rest.client.max-retries";
+  private static final String REST_MAX_CONNECTIONS = "rest.client.max-connections";
+  private static final int REST_MAX_CONNECTIONS_DEFAULT = 100;
+  private static final String REST_MAX_CONNECTIONS_PER_ROUTE = "rest.client.connections-per-route";
+  private static final int REST_MAX_CONNECTIONS_PER_ROUTE_DEFAULT = 100;
```

Comment:
Nit: I guess you could just assign REST_MAX_CONNECTIONS_PER_ROUTE_DEFAULT to REST_MAX_CONNECTIONS_DEFAULT since the default will probably always be 1:1 with the per route default/ but no big deal since I don't expect the defaults to change anyways

Reply:
That's true, but it might be a little confusing so kept separate defaults.